src/views/DiscussionBoard/index.tsx

```diff
@@ -3,41 +3,48 @@ import classNames from 'classnames';
 import { SystemIntakeGRBReviewDiscussionFragment } from 'gql/gen/graphql';
 
 import Alert from 'components/shared/Alert';
+import useDiscussionParams from 'hooks/useDiscussionParams';
 import { DiscussionAlert } from 'types/discussions';
 
 import Discussion from './Discussion';
 import DiscussionModalWrapper from './DiscussionModalWrapper';
+import StartDiscussion from './StartDiscussion';
+import ViewDiscussions from './ViewDiscussions';
 
-// import ViewDiscussions from './ViewDiscussions';
-// import StartDiscussion from './StartDiscussion';
 import './index.scss';
 
 type DiscussionBoardProps = {
   systemIntakeID: string;
   grbDiscussions: SystemIntakeGRBReviewDiscussionFragment[];
-  isOpen: boolean;
-  closeModal: () => void;
 };
 
 function DiscussionBoard({
   systemIntakeID,
-  grbDiscussions,
-  isOpen,
-  closeModal
+  grbDiscussions
 }: DiscussionBoardProps) {
   /** Discussion alert state for form success and error messages */
   const [discussionAlert, setDiscussionAlert] = useState<DiscussionAlert>(null);
 
-  // Get the first discussion from the array for testing purposes
-  const activeDiscussion = grbDiscussions.length > 0 ? grbDiscussions[0] : null;
+  const { getDiscussionParams, pushDiscussionQuery } = useDiscussionParams();
+  const { discussionMode, discussionId } = getDiscussionParams();
+
+  const activeDiscussion =
+    grbDiscussions.find(d => d.initialPost.id === discussionId) || null;
 
   // Reset discussionAlert when side panel is opened or closed
   useEffect(() => {
     setDiscussionAlert(null);
-  }, [setDiscussionAlert, isOpen]);
+  }, [setDiscussionAlert, discussionMode]);
```

Comment:
Thanks for pointing this out

Reply:
I'm actually not sure what should happen here with what you had originally + the new subview states.

The original comment says to reset on open or close, which was before the subview states. I used discussionMode because I figured the concept of open|close was similar enough to the mode subview switching, though the effect now is definitely wrong.

I test reverted to the `isOpen` idea, but the alert just stays up if the user decides to keep interacting with more convos without closing the panel. Not saying a revert is what you're suggesting. Just not sure of where to go with what's set up.

Any ideas on how to approach? @aterstriep

Reply:
Ideally, the `discussionAlert` needs to be reset in two instances:

1. `discussionMode` changes from `reply`
2. `discussionMode` changes from `view`

If I remember right, Mint uses `setTimeout` to close their alerts after a certain time period. So alternatively we could do it that way as well as reset when the modal is opened or closed.

Happy to pair on this later today if that would be helpful!